views/forms/ShareConfigurationForm/index.jsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Typography,
+} from '@mui/material'
+import { yupResolver } from '@hookform/resolvers/yup'
+import * as yup from 'yup'
+
+import ControlledMultiSelect from '../ControlledMultiSelect'
+
+const schema = yup.object().shape({
+  readers: yup.array().when(({ isFixed }) => {
+    if (isFixed === true) yup.required('Owner cannot be removed')
+  }),
+})
+
+const ShareConfigurationForm = ({
+  onClose,
+  onSubmit,
+  open,
+  options,
+  title,
+  formValues,
+}) => {
+  const { control, reset, handleSubmit } = useForm({
+    resolver: yupResolver(schema),
+  })
+  console.log(formValues)
+  useEffect(() => {
+    reset({ readers: formValues })
+  }, [formValues])
```

Comment:
do we still need this?

Reply:
yes, react hook form caches the initial values from the first render, when a user selects a different configuration the cached values have to be updated. 
This is also recommended by the react hook form author